src/fields/ButtonElement.jsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+import Delta from 'quill-delta';
+
+import { adjustColor, textVariablePattern } from '../utils/formHelperFunctions';
+
+const buttonAlignmentMap = {
+    left: 'flex-start',
+    center: 'center',
+    right: 'flex-end'
+};
+
+function ButtonElement({
+    field,
+    fieldValues,
+    conditions,
+    isFilled,
+    displaySteps,
+    submit,
+    hoverStyles = {},
+    selectedStyles = {}
+}) {
+    const [showSpinner, setShowSpinner] = useState(false);
+
+    const elementKey = field.text;
+    const repeat = field.repeat || 0;
+
+    const delta = new Delta(field.text_formatted);
+
+    // TODO (jake): Make this in React
+    const nodes = delta.map((op, i) => {
```

Comment:
you need to decompose logic that's being duplicated

Reply:
Done.